src/main/java/com/kobylchak/bookstore/service/BookServiceImpl.java

```diff
@@ -1,24 +1,40 @@
 package com.kobylchak.bookstore.service;
 
+import com.kobylchak.bookstore.dto.BookDto;
+import com.kobylchak.bookstore.dto.CreateBookRequestDto;
+import com.kobylchak.bookstore.mapper.BookMapper;
 import com.kobylchak.bookstore.model.Book;
 import com.kobylchak.bookstore.repository.BookRepository;
+import jakarta.persistence.EntityNotFoundException;
 import java.util.List;
+import java.util.Optional;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
-
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookRepository.save(bookMapper.toModel(requestDto));
+        return bookMapper.toDto(book);
+    }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getBookById(Long id) {
+        Optional<Book> bookById = bookRepository.findBookById(id);
+        Book book = bookById.orElseThrow(
+                () -> new EntityNotFoundException("Book by id: " + id + "not found"));
```

Comment:
bookById variable is not required
```suggestion
        Book book = bookRepository.findBookById(id).orElseThrow(
                () -> new EntityNotFoundException("Book by id: " + id + "not found"));
```